test/transmuxer.test.js

```diff
@@ -1265,49 +1310,346 @@ QUnit.test('infers sample durations from DTS values', function() {
 
   boxes = mp4.tools.inspect(segment);
   samples = boxes[0].boxes[1].boxes[2].samples;
-  QUnit.equal(samples.length, 3, 'generated three samples');
-  QUnit.equal(samples[0].duration, 1, 'set the first sample duration');
-  QUnit.equal(samples[1].duration, 2, 'set the second sample duration');
-  QUnit.equal(samples[2].duration, 2, 'inferred the final sample duration');
+  QUnit.equal(samples.length, 2, 'generated two samples, filters out pre-IDR');
+  QUnit.equal(samples[0].duration, 3, 'set the first sample duration');
+  QUnit.equal(samples[1].duration, 3, 'set the second sample duration');
 });
 
-QUnit.test('filters pre-IDR samples and caluculate duration correctly', function() {
+QUnit.test('holds onto the last GOP and prepends the subsequent push operation with that GOP', function() {
   var segment, boxes, samples;
+
+  videoSegmentStream.push({
```

Comment:
I had been avoiding this but creating all the test NALs for these tests below seem to distract from the point of the test. What do you think about a helper function of some sort?